### PR TITLE
Login sign up

### PR DESCRIPTION
**Pull request by sainath7878**
Added authorisation and private routes

Files to check: 

- [src/App.js](https://github.com/sainath7878/joddkart/compare/login-signUp?expand=1#diff-3d74dddefb6e35fbffe3c76ec0712d5c416352d9449e2fcc8210a9dee57dff67)
- [src/components/authorization/authorized.jsx](https://github.com/sainath7878/joddkart/compare/login-signUp?expand=1#diff-c904afab8b279678e4ef4446790ec7a326e827d57b30023e87c920a89c6d3f1e)
- [src/components/header/Header.jsx](https://github.com/sainath7878/joddkart/compare/login-signUp?expand=1#diff-837c4f175750b307194e2d05434fb2f5f7a0d7fd3b44c0f22c6ada5938599448)
- [src/components/signIn/signIn.jsx](https://github.com/sainath7878/joddkart/compare/login-signUp?expand=1#diff-681e152a926afb669bd5f4d14dc6552257d4ddc79bb1d36750e7c7bf74150a8f)
- [src/components/signUp/signUp.jsx](https://github.com/sainath7878/joddkart/compare/login-signUp?expand=1#diff-cda0020f3f9d6b1320ea739c6dc3e4b82b163b0475101dfd5dcd9e9462b9583f)
- [src/context/auth-context.js](https://github.com/sainath7878/joddkart/compare/login-signUp?expand=1#diff-50d25f7adf348217c540867a8937453cd2a126a2abdf2681a9e4a838a51f9714)
- [src/reducer/authReducer.js](https://github.com/sainath7878/joddkart/compare/login-signUp?expand=1#diff-775bb0d24e4bc5d5ca19c8dbd21e59d3232fc2b7bcfb42547058db478f086d2d)
- [src/index.js](https://github.com/sainath7878/joddkart/compare/login-signUp?expand=1#diff-bfe9874d239014961b1ae4e89875a6155667db834a410aaaa2ebe3cf89820556)

### Thanks for reviewing!

### CODE REVIEW

File: src/components/productCard/productCard.jsx [hunk 1]
```diff
@@ -3,7 +3,8 @@ import { cartSolid } from "../../assets/images/index";
 import { BiHeartFill, BiXLg, BiStarFill } from "../../assets/icons/Icons";
 import { useProducts } from "../../context/product-context";
 import axios from "axios";
-import { Link } from "react-router-dom";
+import { Link, useNavigate } from "react-router-dom";
+import { useAuth } from "../../context/auth-context";
```
Comment (ApurvaSawant11) [inline src/components/productCard/productCard.jsx]: Try barrel exports.

Check this out: https://blog.justmyrealname.dev/barrel-exporting-in-react

File: src/App.js [hunk 2]
```diff
@@ -6,6 +6,7 @@ import { ProductsListingPage } from "./pages/productsListingPage/productListing"
 import { CartPage } from "./pages/CartPage/CartPage";
 import { WishListPage } from "./pages/wishlistPage/WishListPage"
 import { AuthorizationPage } from "./pages/authorizationPage/authorizationPage"
```
Comment (ApurvaSawant11) [inline src/App.js]: Here also. Try barrel exports

**Review by ApurvaSawant11 — APPROVED** [review]
The code looks good. 

General suggestion:  Try to modularize your code. It will look cleaner and readable

File: src/components/productCard/productCard.jsx (hunk 1 above)
Comment (sainath7878) [reply src/components/productCard/productCard.jsx]: Done. Thanks for pointing out!

File: src/App.js (hunk 2 above)
Comment (sainath7878) [reply src/App.js]: Done!

File: src/components/authorization/authorized.jsx [hunk 3]
```diff
@@ -0,0 +1,12 @@
+import { Navigate, Outlet, useLocation } from "react-router-dom";
+import { useAuth } from "../../context/auth-context";
+
+function Authorized() {
+  const { authState } = useAuth();
+  const location = useLocation();
+  return (
+    <>{authState.isLoggedIn ? <Outlet /> : <Navigate to="/signin" from={{state: location}} replace />}</>
+  );
+}
```
Comment (anik31) [inline src/components/authorization/authorized.jsx:10]: appreciation - good that you have created a separate component for authorization

File: src/context/auth-context.js [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+const AuthContext = createContext();
+
+
+function AuthProvider({ children }) {
+
+    const [authState, authDispatch] = useReducer(authReducer, initialAuthState)
+    return (
+        <AuthContext.Provider value={{ authState, authDispatch }}>
+            {children}
+        </ AuthContext.Provider >
```
Comment (anik31) [inline src/context/auth-context.js:13]: appreciation - good that you're using a separate context for auth

**Review by anik31 — APPROVED** [review]
Good work @sainath7878 , the functionality is working perfectly in preview & the code is also clean

**Pull request merged**